src/transfers.rs

```diff
@@ -146,3 +152,287 @@ pub fn generates_transfers(
     collections.transfers = Collection::new(new_transfers);
     Model::new(collections)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{generate_transfers_from_sp, generates_transfers, make_transfers_map, TransferMap};
+    use crate::{
+        model::Model,
+        objects::{Coord, Time, Transfer},
+        ModelBuilder,
+    };
+    use typed_index_collection::Collection;
+
+    // A - B    92
+    // A - C    158
+    // B - C    66
+    fn base_model() -> Model {
+        let model_builder = ModelBuilder::default();
+        let transit_model = model_builder
+            .vj("vj1", |vj_builder| {
+                vj_builder
+                    .route("route1")
+                    .st("A", "10:00:00")
+                    .st("B", "11:00:00")
+                    .st("C", "12:00:00");
+            })
+            .add_transfer("A", "B", Time::new(0, 10, 0))
+            .build();
+
+        let mut collections = transit_model.into_collections();
+        collections.stop_points.get_mut("A").unwrap().coord =
+            Coord::from(("2.38951".to_string(), "48.852245".to_string()));
+        collections.stop_points.get_mut("B").unwrap().coord =
+            Coord::from(("2.390403".to_string(), "48.85165".to_string()));
+        collections.stop_points.get_mut("C").unwrap().coord =
+            Coord::from(("2.390403".to_string(), "48.85165".to_string()));
+
+        Model::new(collections).unwrap()
+    }
+
+    #[test]
+    // fn test_generate_transfers_from_sp() {
+    fn test_make_transfers_map() {
+        let model = base_model();
+        let transfers_map = make_transfers_map(model.transfers.clone(), &model.stop_points);
+
+        let mut expected = TransferMap::new();
+        let transfer_a_b = Transfer {
+            from_stop_id: "A".to_string(),
+            to_stop_id: "B".to_string(),
+            min_transfer_time: Some(8),
+            real_min_transfer_time: Some(10),
+            equipment_id: None,
+        };
+        expected.insert(
+            (
+                model.stop_points.get_idx("A").unwrap(),
+                model.stop_points.get_idx("B").unwrap(),
+            ),
+            transfer_a_b,
+        );
```

Comment:
ℹ️ you can use a scope (if you want)
```suggestion
        let expected = {
            let map = TransferMap::new();
            let transfer_a_b = Transfer {
                from_stop_id: "A".to_string(),
                to_stop_id: "B".to_string(),
                min_transfer_time: Some(8),
                real_min_transfer_time: Some(10),
                equipment_id: None,
            };
            expected.insert(
                (
                    model.stop_points.get_idx("A").unwrap(),
                    model.stop_points.get_idx("B").unwrap(),
                ),
                transfer_a_b,
            );
            map
        };
```